### PR TITLE
Backtracking Algorithms

### PR DESCRIPTION
**Pull request by SubhradeepSS**
@Bhavnaharitsa please review my PR for the issue I created #66

### CODE REVIEW

File: Algorithms/Backtracking/N Queen Problem.cpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+		} 
+	} 
+
+	/* If the queen cannot be placed in any row in 
+		this colum col then return false */
+	return false; 
+} 
+
+/* This function solves the N Queen problem using 
+Backtracking. It mainly uses solveNQUtil() to
```
Comment (Bhavnaharitsa) [inline Algorithms/Backtracking/N Queen Problem.cpp]: No need of this comments

File: Algorithms/Backtracking/N Queen Problem.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+// driver program to test above function 
+int main() 
+{ 
+	solveNQ(); 
+	return 0; 
+} 
+
+
+Output:
```
Comment (Bhavnaharitsa) [inline Algorithms/Backtracking/N Queen Problem.cpp]: Comment output

File: Algorithms/Backtracking/N Queen Problem.cpp [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+		for (int j = 0; j < N; j++) 
+			printf(" %d ", board[i][j]); 
+		printf("\n"); 
+	} 
+} 
+
+/* A utility function to check if a queen can 
+be placed on board[row][col]. Note that this 
+function is called when "col" queens are 
+already placed in columns from 0 to col -1.
```
Comment (Bhavnaharitsa) [inline Algorithms/Backtracking/N Queen Problem.cpp]: Paste it on top with name of algorithm and description

**Comment by SubhradeepSS** [comment]
@Bhavnaharitsa  updated.
Please suggest further changes.

File: Algorithms/Backtracking/N Queen Problem.cpp (hunk 3 above)
Comment (SubhradeepSS) [reply Algorithms/Backtracking/N Queen Problem.cpp]: @Bhavnaharitsa please review.
Any further changes required?

File: Algorithms/Backtracking/Hamiltonian Cycle.cpp [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+	// Print the solution 
+	hamCycle(graph2); 
+
+	return 0; 
+} 
+
+
+Output:
+Solution Exists: Following is one Hamiltonian Cycle
```
Comment (Bhavnaharitsa) [inline Algorithms/Backtracking/Hamiltonian Cycle.cpp:153]: Comment output

File: Algorithms/Backtracking/M Coloring Problem.cpp [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+		} 
+	} 
+
+	/* If no color can be assigned to this vertex then return false */
+	return false; 
+} 
+
+/* This function solves the m Coloring problem using Backtracking. 
+It mainly uses graphColoringUtil() to solve the problem. It returns 
+false if the m colors cannot be assigned, otherwise return true and
```
Comment (Bhavnaharitsa) [inline Algorithms/Backtracking/M Coloring Problem.cpp]: no need explanation

File: Algorithms/Backtracking/M Coloring Problem.cpp [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+		{1, 1, 0, 1}, 
+		{1, 0, 1, 0}, 
+	}; 
+	int m = 3; // Number of colors 
+	graphColoring (graph, m); 
+	return 0; 
+} 
+
+
+Output:
```
Comment (Bhavnaharitsa) [inline Algorithms/Backtracking/M Coloring Problem.cpp]: comment output

Comment (SubhradeepSS) [reply Algorithms/Backtracking/M Coloring Problem.cpp]: Output is already in comment section at the end of the code

File: Algorithms/Backtracking/Hamiltonian Cycle.cpp (hunk 4 above)
Comment (SubhradeepSS) [reply Algorithms/Backtracking/Hamiltonian Cycle.cpp:153]: Output is already in comment section at the end of the code

**Comment by SubhradeepSS** [comment]
@Bhavnaharitsa  please review the new PR #67 ,not the old one

**Comment by Bhavnaharitsa** [comment]
I am on pr #67 
Which one is the old one?

**Comment by SubhradeepSS** [comment]
> I am on pr #67
> Which one is the old one?

Output is already commented out

**Comment by SubhradeepSS** [comment]
Issue #71 fixed
Please review @Bhavnaharitsa

File: Algorithms/Backtracking/N Queen Problem.cpp (hunk 2 above)
Comment (Bhavnaharitsa) [reply Algorithms/Backtracking/N Queen Problem.cpp]: Where is it? It is not

File: Algorithms/Backtracking/N Queen Problem.cpp (hunk 3 above)
Comment (Bhavnaharitsa) [reply Algorithms/Backtracking/N Queen Problem.cpp]: Do the changes mentioned below

**Comment by SubhradeepSS** [comment]
@Bhavnaharitsa  I think there is some issue;I updated the things on my repo.But its not showing.
I am closing this PR and opening a new one

**Comment by SubhradeepSS** [comment]
> @Bhavnaharitsa I think there is some issue;I updated the things on my repo.But its not showing.
> I am closing this PR and opening a new one

@Bhavnaharitsa I squashed the earlier commits in which output was not commented.
U can check out in my master branch that I have updated

**Comment by Bhavnaharitsa** [comment]
Conflicts

**Comment by SubhradeepSS** [comment]
> Conflicts

Solved

**Pull request merged**